src/test/main.cpp

```diff
@@ -546,6 +546,52 @@ void interpreter_tests(Test::Tester& tests) {
 		    return ExitStatusTag::Ok;
 	    }}));
 
+	tests.add_test(std::make_unique<Test::InterpreterTestSet>(
+	    R"(
+			a := c;
+			b := a.id { 10 };
+			c := union { id : int(<>); }(<>);
+			__invoke := fn() => 0;
+		)",
+	    Testers {+[](Interpreter::Interpreter& env) -> ExitStatusTag {
+		    return Assert::equals(eval_expression("__invoke()", env), 0);
+	    }}));
+
+	tests.add_test(std::make_unique<Test::InterpreterTestSet>(
+	    R"(
+			a := c;
+			b := a.id { 10 };
+			c := union { id : int(<>); }(<>);
+			__invoke := fn() => 0;
+		)",
+	    Testers {+[](Interpreter::Interpreter& env) -> ExitStatusTag {
+		    return Assert::equals(eval_expression("__invoke()", env), 0);
+	    }}));
+
+	tests.add_test(std::make_unique<Test::InterpreterTestSet>(
+	    R"jp(
+			b := union {
+				id : string(<>);
+				num : int(<>);
+			}(<>);
+
+			serialize := fn ( x ) {
+				y := match(x) {
+					id { s } => s;
+					num { n } => "(number)";
+				};
+				return y;
+			};
+
+			__invoke := fn() {
+				print(serialize(b.id { "xxx" }));
+				print(serialize(b.num { 4242 }));
+			};
+		)jp",
+	    Testers {+[](Interpreter::Interpreter& env) -> ExitStatusTag {
+		    return Assert::equals(eval_expression("__invoke()", env), 3);
+	    }}));
```

Comment:
really weird how it doesn't fail, huh

Reply:
about `R"jp()jp"`, https://en.cppreference.com/w/cpp/language/string_literal

raw string literals let you use a specific delimiter, in case the default one appears within the string itself.

Reply:
fixed